src/Analyzers/Core/CodeFixes/RemoveUnnecessarySuppressions/RemoveLegacySuppressionsCodeFixProvider.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.RemoveUnnecessarySuppressions
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = PredefinedCodeFixProviderNames.RemoveUnnecessarySuppressions), Shared]
+    internal sealed class RemoveLegacySuppressionsCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public RemoveLegacySuppressionsCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.LegacyFormatSuppressMessageAttributeDiagnosticId);
+
+        internal override CodeFixCategory CodeFixCategory
+            => CodeFixCategory.CodeQuality;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                if (diagnostic.Properties?.ContainsKey(AbstractRemoveUnnecessarySuppressionsDiagnosticAnalyzer.DocCommentIdKey) == true)
+                {
+                    context.RegisterCodeFix(
+                        new MyCodeAction(c => FixAsync(context.Document, diagnostic, c)),
+                        diagnostic);
+                }
+            }
+
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = editor.OriginalRoot.FindNode(diagnostic.Location.SourceSpan);
+                if (node == null)
```

Comment:
these checks are weird.  it means you may get a fix offered that does nothing.  if you're going to bail, it should be in RegisterCodeFixesAsync

Reply:
Sure, let me move the check.